src/core/ColorPicker/ColorInputPanel.tsx

```diff
@@ -65,18 +65,22 @@ export const ColorInputPanel = React.forwardRef(
       if (currentFormat === 'hsl') {
         const hsl = activeColor.toHslColor();
         setInput([
-          hsvColor.h.toString(), // use hsvColor to preserve hue for 0,0,0 edge case
-          hsl.s.toString(),
-          hsl.l.toString(),
-          hsl.a?.toFixed(2) ?? (activeColor.getAlpha() / 255).toFixed(2),
+          ColorValue.getFormattedColorNumber(hsvColor.h), // use hsvColor to preserve hue for 0,0,0 edge case
+          ColorValue.getFormattedColorNumber(hsl.s),
+          ColorValue.getFormattedColorNumber(hsl.l),
+          ColorValue.getFormattedColorNumber(
```

Comment:
The input is messed up, on certain colors you can't input the hsl value that you want because it changes it to random decimal values. Also note how the S value also changes slightly when inputting the L value. I think this has to do with all the converting back and forth
![input](https://user-images.githubusercontent.com/88331924/141148680-7e606016-f297-423d-97bc-015060cc8c10.gif)

Reply:
This is a by-product of using RGB (tbgr) as the single source of truth. But I want to keep that as it is the cornerstone of all colors in imodeljs. If you go from rgb -> hsl -> rgb then there is no loss but going from hsl -> rgb -> hsl is not foolproof. Same as before but at least now you can do the rgb -> hsl -> rgb.

Reply:
I think it would help to save the entire original input instead of just `_hue`

Reply:
@saaaaaally @mayank99 - just pushed updates that I believe improves the results -- I cleaned the hsv conversion - but there is still some changing just due to hsl -> rgb -> hsl conversion.

Reply:
Lets move forward with this then. We can think about saving original input in the future.

Reply:
Created https://github.com/iTwin/iTwinUI-react/issues/429#issue-1052036750 to address this last remaining bug